Adds 'Backup, high availability, and resilience tools' landing page (#301)
elastic/docs-content: deploy-manage/tools.md

Reviewer: are you still planning on changing this url?

Author: Probably yes, but I’d prefer to create a new issue and PR for it to ensure that all links in other content remain safe. I think we can treat this as a secondary priority for now until we complete the base migration work - what do you think?

Reviewer: sure

---
mapped_pages:
  - https://www.elastic.co/guide/en/elasticsearch/reference/current/high-availability.html

applies:
  stack: all

>   serverless: all

Reviewer: please remove serverless from here because none of the tools are available for serverless

```suggestion
```

> 

Reviewer: please add the `applies` tag with the relevant products inside of that front matter block 

https://elastic.github.io/docs-builder/syntax/applies.html

# Backup, high availability, and resilience tools [high-availability]

> Elasticsearch provides comprehensive tools to safeguard data, ensure continuous availability, and maintain resilience. These tools are designed to support disaster recovery strategies, enabling businesses to protect critical information and minimize downtime, and maintain high availability in case of unexpected failures.

Reviewer: suggest always having a "in this section, you'll learn" statement for overview pages

```suggestion
Elastic provides comprehensive tools to safeguard data, ensure continuous availability, and maintain resilience. These tools are designed to support disaster recovery strategies, enabling businesses to protect critical information and minimize downtime, and maintain high availability in case of unexpected failures. In this section, you'll learn about these tools and how to implement them in your environment.
```

For strategies to design resilient clusters, see **[Availability and resilience](production-guidance/availability-and-resilience.md)**.

> ::::{note} 
> Snapshot and restore, as well as cross-cluster replication, are not currently available in serverless deployments, but they are planned for future releases. For more information, see **[LINK-TODO]**.
> ::::

Reviewer: - prefer `Elastic Cloud Serverless` when you have the room to spell it out
- Elastic Cloud Serverless uses `projects`, not `deployments`.
- ECS also doesn't have "releases" per se (it's versionless), so we would just say "future"

provided an anchor to link down right to the feature plans for serverless

```suggestion
::::{note} 
The snapshot and restore and cross-cluster replication features are currently not available for Elastic Cloud Serverless projects. These features will be introduced in the future. For more information, refer to [Serverless differences](/deploy-manage/deploy/elastic-cloud/differences-from-other-elasticsearch-offerings.md#elasticsearch-differences-serverless-feature-categories).
::::
```

> ## Snapshots

Reviewer: given that this feature has been referred to as "snapshot and restore" historically, let's use the full name for the title

```suggestion
## Snapshot and restore
```

Snapshots in Elasticsearch are point-in-time backups that include your cluster's data, settings, and overall state. They capture all the information necessary to restore your cluster to a specific moment in time, making them essential for protecting data, recovering from unexpected issues, and transferring data between clusters. Snapshots are a reliable way to ensure the safety of your data and maintain the continuity of your operations.

You can perform the following tasks to manage snapshots and snapshot repositories:

- **[Register a repository](tools/snapshot-and-restore/manage-snapshot-repositories.md):** Configure storage repositories (for example, S3, Azure, Google Cloud) to store snapshots. The way that you register repositories differs depending on your deployment method:

>   - **[Elastic Cloud Hosted](tools/snapshot-and-restore/elastic-cloud-hosted.md):** Deployments come with a preconfigured S3 repository for automatic backups, simplifying the setup process. You can also register external repositories, such as Azure, and Google Cloud, for more flexibility.

Reviewer: ```suggestion
  - **[Elastic Cloud Hosted](tools/snapshot-and-restore/elastic-cloud-hosted.md):** Deployments come with a preconfigured S3 repository for automatic backups, simplifying the setup process. You can also register external repositories, such as Azure and Google Cloud, for more flexibility.
```

  - **[Elastic Cloud Enterprise](tools/snapshot-and-restore/cloud-enterprise.md):** Repository configuration is managed through the Elastic Cloud Enterprise user interface and automatically linked to deployments.
  - **[Elastic Cloud on Kubernetes](tools/snapshot-and-restore/cloud-on-k8s.md) and [self-managed](tools/snapshot-and-restore/self-managed.md) deployments:** Repositories must be configured manually.

- **[Create snapshots](tools/snapshot-and-restore/create-snapshots.md):** Manually or automatically create backups of your cluster.
- **[Restore a snapshot](tools/snapshot-and-restore/restore-snapshot.md):** Recover indices, data streams, or the entire cluster to revert to a previous state. You can choose to restore specific parts of a snapshot, such as a single index, or perform a full restore.

To reduce storage costs for infrequently accessed data while maintaining access, you can also create **[searchable snapshots](tools/snapshot-and-restore/searchable-snapshots.md)**.

> ::::{note}
> Snapshot configurations vary across Elastic Cloud Hosted, Elastic Cloud Enterprise (ECE), Elastic Cloud on Kubernetes (ECK), and self-managed deployments.
> ::::

Reviewer: This note is duplicative so we can remove it 

```suggestion
```

Reviewer 2: Totally agreed, I think we should remove it except if there was any specific need (and in such case the note would need to explain its point, as it's currently vague).

## Cross-cluster replication (CCR)

> **[Cross-cluster replication (CCR)](tools/cross-cluster-replication.md)** is a feature in Elasticsearch that allows you to replicate data in real time from a leader cluster to one or more follower clusters. This replication ensures that data is synchronized across clusters, providing continuity, redundancy, and enhanced data accessibility. 

Reviewer: will this content show up on a child page? 

>    To effectively distribute read and write operations across nodes, the nodes in a cluster need good, reliable connections to each other. To provide better connections, you typically co-locate the nodes in the same data center or nearby data centers.
>
>    Co-locating nodes in a single location exposes you to the risk of a single outage taking your entire cluster offline. To maintain high availability, you can prepare a second cluster that can take over in case of disaster by implementing {{ccr}} (CCR).

Author: Yes, I was planning to move this to the CCR landing page.

> **[Cross-cluster replication (CCR)](tools/cross-cluster-replication.md)** is a feature in Elasticsearch that allows you to replicate data in real time from a leader cluster to one or more follower clusters. This replication ensures that data is synchronized across clusters, providing continuity, redundancy, and enhanced data accessibility. 
> 
> CCR provides a way to automatically synchronize indices from a leader cluster to a follower cluster. This cluster could be in a different data center or even a different continent from the leader cluster. If the primary cluster fails, the secondary cluster can take over.

Reviewer: The second para feels a lot like the one above it. let's blend them

```suggestion
**[Cross-cluster replication (CCR)](tools/cross-cluster-replication.md)** is a feature in Elasticsearch that allows you to replicate data in real time from a leader cluster to one or more follower clusters. This cluster could be in a different data center or even a different continent from the leader cluster. 

This replication ensures that data is synchronized across clusters, providing continuity, redundancy, and enhanced data accessibility.  For example, if the primary cluster fails, the secondary cluster can take over.
```

::::{note}

> CCR relies on **[Remote clusters](remote-clusters.md)** functionality to establish and manage connections between the leader and the follower clusters.

Reviewer: you can use sentence case unless you need to refer to the name of the topic specifically (like "Refer to [Remote clusters]")

```suggestion
CCR relies on **[remote clusters](remote-clusters.md)** functionality to establish and manage connections between the leader and the follower clusters.
```

::::

You can perform the following tasks to manage cross-cluster replication:

> - **[Set up CCR](tools/cross-cluster-replication/set-up-cross-cluster-replication.md):** Configure leader and follower clusters for data replication. To enable CCR, you need to configure **[Remote clusters](remote-clusters.md)**.

Reviewer: I don't think the prereq needs to be stated here. you have it in the note above, and we can allow the setup topic to cover any prerequisites.

```suggestion
- **[Set up CCR](tools/cross-cluster-replication/set-up-cross-cluster-replication.md):** Configure leader and follower clusters for data replication.
```

- **[Manage CCR](tools/cross-cluster-replication/manage-cross-cluster-replication.md):** Monitor and manage replicated indices.
- **[Automate replication](tools/cross-cluster-replication/manage-auto-follow-patterns.md):** Use auto-follow patterns to automatically replicate newly created indices.

> - **Failover clusters:** Configure **[uni-directional](tools/cross-cluster-replication/uni-directional-disaster-recovery.md)** or **[bi-directional](tools/cross-cluster-replication/bi-directional-disaster-recovery.md)** CCR for redundancy and disaster recovery.

Reviewer: for parallelism, this should be phrased with a verb

```suggestion
- **Set up failover clusters:** Configure **[uni-directional](tools/cross-cluster-replication/uni-directional-disaster-recovery.md)** or **[bi-directional](tools/cross-cluster-replication/bi-directional-disaster-recovery.md)** CCR for redundancy and disaster recovery.
```

> - **[Cluster upgrade considerations when using CCR](upgrade.md):** Maintain cluster performance and security by upgrading your Elasticsearch clusters. Use the tools and procedures suited to your deployment type: Elastic Cloud Hosted, ECE, ECK, or self-managed.

Reviewer: This description does not match the text in the [page](https://www.elastic.co/guide/en/elasticsearch/reference/current/ccr-upgrading.html) I think you're referencing. 

```suggestion
- **[Review cluster upgrade considerations when using CCR](upgrade.md):** If you're using CCR, then you might need to upgrade your clusters in a specific order to prevent errors. Review the considerations and recommended procedures for performing upgrades on CCR leaders and followers.
```

Reviewer: @jmikell821 fyi we have this little upgrade-related topic buried over here. you might want to reference it in your docs
